File: Desktop/Code/NewFolder/6.2/stack.cpp
#include "stack.h"
#include <stdio.h>
#include <cstdlib>
#include <iostream>

struct StackElement {
    int value;
    StackElement *next;
};

struct Stack {
    StackElement *head; 
};

Stack *createStack() {
    return new Stack{};
}

void deleteStack(Stack *stack) {
    while (!isEmpty(stack)) { 
        StackElement *temp = stack->head;
        stack->head = stack->head->next;
        delete temp;
    }
    delete stack;
}

bool isEmpty (Stack *stack) {
    return !stack->head;
}

void push(Stack *stack, int value) {
    StackElement* newElement = new StackElement;
    newElement->value = value;
    newElement->next = stack->head;
    stack->head = newElement;
}

int pop(Stack *stack) {
    StackElement* newHead = stack->head->next;
    int val = stack->head->value;
    delete stack->head;
    stack->head = newHead;
    return val;
}

File: 2Semester/2homework/2.1/2.1/Stack.cs
using System;

namespace StackRealization
{
    public class Stack
    {
        private class StackElement
        {
            public int Value { get; set; }
            public StackElement Next { get; set; }
        }

            private StackElement head = null;
        public void Push(int value)
        {
            this.head = new StackElement
            {
                Next = this.head,
                Value = value
            };
        }

        public int Pop()
        {
            if (this.Empty)
            {
                return -1;
            }
            var value = this.head.Value;
            this.head = this.head.Next;
            return value;
        }

        public bool Empty => this.head == null;
    }
}